tools/ci_connector_ops/ci_connector_ops/pipelines/connectors_ci.py

```diff
@@ -0,0 +1,154 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+import json
+import logging
+import os
+import sys
+from pathlib import Path
+from typing import List, Optional
+
+import anyio
+import click
+import dagger
+from ci_connector_ops.pipelines.actions import build_contexts, builds, tests
+from ci_connector_ops.pipelines.utils import StepStatus, write_connector_secrets_to_local_storage
+from ci_connector_ops.utils import Connector, ConnectorLanguage, get_changed_connectors
+from dagger.api.gen import Client, Container
+from graphql import GraphQLError
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+async def run_connector_test_pipelines(dagger_client: Client, connector: Connector, gsm_credentials: Optional[str]):
```

Comment:
Since this is async, can we run any of the sub-tasks in parallel?

Reply:
Yes we can :)